Add support for swift test

gfontenot opened the pull request:
Swift Package Manager ships a test runner that is accessible via `swift test`.
This commit adds support for working with `swift test` via vim-test.

There's definitely some fragility in here still that might need to get worked
out over time. Off the top of my head, some potential issues:

- Users might locate their tests in a directory named something other than
  "Tests". Doing so would break the test file detection (and the module name
  parsing).
- If users use a test lib like Quick, this won't properly parse module info
  (and filtering Quick tests with swift test doesn't work at all).

That being said, I think this is a good start, and getting this into the
plugin would mean being able to deploy fixes for these issues by tweaking the
regexes instead of needing to write this whole thing from scratch.

gfontenot commented:
I'm not sure why CI is failing here... I can run these tests locally and see them pass, and I can also run the plugin against actual code and see it perform as expected 😕

codeinabox commented on autoload/test/swift/swiftpm.vim:
```diff
@@ -0,0 +1,67 @@
+endfunction
+
+function! s:parse_nearest_test_info(position)
+  let l:info = g:test#base#nearest_test(a:position, g:test#swift#patterns)
```
I think you need to put this instead judging by the error

```viml
let l:info = test#base#nearest_test(a:position, g:test#swift#patterns)
```

gfontenot commented:
Fixed CI and added Swift support to the README. I didn't see anywhere obvious to update `doc/test.txt`. Happy to do that if you point me to what should be updated.

codeinabox commented:
For doc/test.txt it's just a case of adding something like this amongst all the runner commands

```
                                                *test-: SwiftPM*
:SwiftPM [args]               Uses the `swift test` command.
```

gfontenot commented:
Ah! My bad. Added that there now. I don't know why I didn't see that section.

janko-m commented:
Looks great, thank you very much!
